### DISCUSSION TITLE
ListInstancesAsync not returning all instances when InstanceIdPrefix is passed

### DISCUSSION THREAD

**Discussion by bhuvesh321** [Azure/azure-functions-durable-extension#2011] [Q&A]
We are using Azure durable functions with custom deployment on service fabric servers. We are using same instanceId prefix for parent orchestration and suborchestrations but while querying the table using ListInstancesAsync with InstanceIdPrefix, its returning only 1 parent orchestration instance. Other sub orchestration instances are not returned. Can you please help?

More details:-
In the attached screenshot, there are multiple instances with same prefix. Only 1 record is returned after passing 2 below values in OrchestrationStatusQueryCondition.
InstanceIdPrefix - 0abb6ebb-d712-453a-97c4-6c7c1f78f49f
TaskHubNames - <Task hub name>
OrchestrationStatusQueryResult results = await orchestrationClient.ListInstancesAsync(queryCondition, cancellationToken).ConfigureAwait(false);

I have recently upgraded to latest DTF version 2.6.0. Please let me know if you need more information.
![DTFInstances](https://user-images.githubusercontent.com/30367990/142178432-3aa4ee41-391a-4109-ae09-df9af74b14cf.png)

**Comment by cgillum** [comment]
Can you share the code you wrote that generates the queryCondition? I was not able to reproduce this problem, even if I use your exact same InstanceIdPrefix value (I got all 5 instances back).

**Reply by bhuvesh321** [reply]
Thanks Chris for looking into this... The issue which I found is that if I pass the TaskHubNames then only 1 record is returned however if I pass the TaskHubName as null then 5 records are returned. The task hub name I passed is correct. Please find the code below:


                queryCondition.CreatedTimeFrom = DateTime.UtcNow.AddDays(-ageInDays);
                queryCondition.CreatedTimeTo = DateTime.UtcNow;
                queryCondition.TaskHubNames = new List<string>() { TaskHubSimulatioName };
                queryCondition.InstanceIdPrefix = "0abb6ebb-d712-453a-97c4-6c7c1f78f49f";
                string continuationToken = string.Empty;
                queryCondition.ContinuationToken = continuationToken;
                logger.LogInformation(LogHelper.GetMessage($"AutoHealRunner: Getting failed instances for last {ageInDays}. From:{queryCondition.CreatedTimeFrom}, To:{queryCondition.CreatedTimeTo}"));
                List<DurableOrchestrationStatus> instances = new List<DurableOrchestrationStatus>();
                do
                {
                    OrchestrationStatusQueryResult results = await orchestrationClient.ListInstancesAsync(queryCondition, cancellationToken).ConfigureAwait(false);
                    instances.AddRange(results.DurableOrchestrationState);

![image](https://user-images.githubusercontent.com/30367990/142580601-b9302ef7-1939-4e06-9825-436c911d838c.png)

**Reply by cgillum** [reply]
Take a look at the _Instances_ table in your Azure Storage account (it should be named like _{taskHubName}Instances_). The PartitionKey field is the orchestration instance ID. For these specific orchestration instances, do they all have a TaskHubName property that matches the value for `TaskHubSimulationName`? This is what `queryCondition.TaskHubNames` should be querying against.  If they all have the correct value, then I wonder if there may be a bug in the Azure Storage query that gets generated.
